core/phase.js

```diff
@@ -206,6 +234,8 @@ PhaseBaseRuntime.prototype.put = function(data, tags) {
   } else {
     this.tags = this.baseTags.clone();
   }
+  // TODO: This misses tags when they are set after calling put().
+  trace.enabled && flowItemPut(this, this.tags.tags);
```

Comment:
why aren't the other invocations (t.start, flowItemGet) guarded by trace.enabled too?